src/dist/manifest.rs

```diff
@@ -496,6 +496,24 @@ impl Component {
             is_extension,
         }
     }
+
+    pub fn new_with_target(pkg_with_target: &str, is_extension: bool) -> Option<Self> {
+        for (pos, _) in pkg_with_target.match_indices('-') {
+            let pkg = &pkg_with_target[0..pos];
+            let target = &pkg_with_target[pos + 1..];
+            if let Some(partial) = PartialTargetTriple::new(target) {
+                if partial.arch.is_some() && partial.os.is_some() && partial.env.is_some() {
```

Comment:
This specific bit feels like reaching into the guts of `PartialTargetTriple`. Without knowing *any* of the context, I'd have expected this to be a method like `partial.is_valid()`

Reply:
Maybe even something like `fn validate(self) -> Option<TargetTriple>`?

Reply:
Partials are valid if any of the parts are available, I'm kinda using it a bit iffily since I want a full triple.

I've added a `impl std::convert::TryFrom<PartialTargetTriple> for TargetTriple` and I'm using that.  Thanks for the idea.